### PR TITLE
Navnoor TLI: Refactor Password Change, Login, and Registration Logic in Use Cases

### PR DESCRIPTION
**Pull request by Navnoorsinghmahal**
### Changes Made
- Added docstrings with return type annotations to the interactor methods for clarity.

### CODE REVIEW

File: backend/app/use_cases/change_password_interactor.py [hunk 1]
```diff
@@ -1,10 +1,27 @@
...
     def __init__(self, user_repo: UserRepository):
+        """
+        Initializes the ChangePasswordInteractor with a user repository.
+
+        :param user_repo: Instance of UserRepository for user data operations.
+        """
         self.user_repo = user_repo
 
-    def execute(self, email, old_password, new_password):
+    def execute(self, email, old_password, new_password)-> dict:
```
Comment (Jai0212) [inline backend/app/use_cases/change_password_interactor.py:15]: add type annotations for the parameters also like email: str and so on

**Review by Jai0212 — CHANGES_REQUESTED** [review]
docstrings are fine, see comment about parameter annotations

**Review by Jai0212 — COMMENTED** [review]
Looks good

**Review by Jai0212 — APPROVED** [review]

**Comment by Jai0212** [comment]
backend/app/controllers/__pycache__/app.cpython-312.pyc

can u remove this file and commit again, it will help solve merge conflict

**Comment by Navnoorsinghmahal** [comment]
> backend/app/controllers/**pycache**/app.cpython-312.pyc
> 
> can u remove this file and commit again, it will help solve merge conflict

Can you try to merge now

**Comment by Jai0212** [comment]
> > backend/app/controllers/**pycache**/app.cpython-312.pyc
> > can u remove this file and commit again, it will help solve merge conflict
> 
> Can you try to merge now

push also

**Pull request merged**